index.d.ts

```diff
@@ -52,15 +43,18 @@ export interface AnyAction extends Action {
  *
  * *Do not put API calls into reducers.*
  *
- * @template S State object type.
+ * @template S The type of state consumed and produced by this reducer.
+ * @template A The type of actions the reducer can potentially respond to.
  */
-export type Reducer<S> = (state: S, action: AnyAction) => S;
+export type Reducer<S = {}, A extends Action = Action> = (state: S, action: A) => S;
```

Comment:
You could certainly opt to include the init action in your top-level union of action types if you want to capture absolutely all actions that might be passing through your reducer. The point is to constrain the types to some known, closed universe so that you can have type safety and semantic awareness for refactoring.